shared/modules/network.utils.ts

```diff
@@ -91,3 +103,50 @@ export function convertNetworkId(value: unknown): string | null {
   }
   return null;
 }
+
+/**
+ * Convert an eip155 CAIP chain ID to a hex chain ID.
+ *
+ * @param id - The CAIP chain ID to convert.
+ * @returns The hex chain ID.
+ */
+export function convertCaipToHexChainId(id: CaipChainId): Hex {
+  const { namespace, reference } = parseCaipChainId(id);
+  if (namespace === KnownCaipNamespace.Eip155) {
+    return `0x${parseInt(reference, 10).toString(16)}`;
+  }
+
+  throw new Error(
+    `Unsupported CAIP chain ID namespace: ${namespace}. Only eip155 is supported.`,
+  );
+}
+
+/**
+ * Sorts a list of networks based on the order of their chain IDs.
+ *
+ * @param networks - The networks to sort.
+ * @param sortedChainIds - The chain IDs to sort by.
+ * @returns The sorted list of networks.
+ */
+export const sortNetworks = (
+  networks: Record<string, MultichainNetworkConfiguration>,
+  sortedChainIds: { networkId: string }[],
+): MultichainNetworkConfiguration[] =>
+  Object.values(networks).sort(
+    (a, b) =>
+      sortedChainIds.findIndex(({ networkId }) => networkId === a.chainId) -
+      sortedChainIds.findIndex(({ networkId }) => networkId === b.chainId),
+  );
+
+/**
+ * Get the network icon for the given chain ID.
+ *
+ * @param chainId - The chain ID to get the icon for in Caip-19 format.
+ * @param isEvm - Whether the chain ID is for an EVM network.
+ * @returns The URL of the network icon.
+ */
+export const getNetworkIcon = (chainId: CaipChainId, isEvm: boolean) => {
```

Comment:
I don't think the `chainId` and `isEvm` should be separated from each other here.

We could probably take a `MultichainNetworkConfiguration` as parameter and use `isEvm` and `chainId` from this config.

My reasoning is, if a caller calls `getNetworkIcon('bip122:...', true)`, then the function will throw because of `convertCaipToHexChainId`.

Reply:
good point -  done in https://github.com/MetaMask/metamask-extension/pull/30433/commits/13233a7b36e4680b623534cbbca14ca6859c6c53